### PR TITLE
bpo-32891: rename numbers.Integral to numbers.Integer

### PR DESCRIPTION
**Pull request by skirpichev**
I'm not sure how to reproduce pickle problem, mentioned by Serhiy in the issue thread.  Seems to be working...

There are few things that might be changed too, but I'm not sure they are suitable for this pr:
1. pymath.h's macroses (e.g. `_Py_InIntegralTypeRange` -> `_Py_InIntegerTypeRange`)
2. Few Decimal's methods (e.g. `to_integral()` -> `to_integer()`)

https://bugs.python.org/issue32891

### CODE REVIEW

**Comment by skirpichev** [comment]
@terryjreedy, I would appreciate your review, as issue creator.  The patch itself is tiny (Lib/numbers.py, Doc/library/numbers.rst and few places where the new class is referenced, instead of the old).  The rest is my attempt to make docs more consistent in using integer vs integral.  I can split that part to a separate commit, if this simplify review somehow.

BTW, [PEP 3141](https://www.python.org/dev/peps/pep-3141/) references the old naming scheme.  Perhaps, it must be updated?

File: Lib/random.py [hunk 1]
```diff
@@ -719,7 +719,7 @@ def gammavariate(self, alpha, beta):
         if alpha > 1.0:
 
             # Uses R.C.H. Cheng, "The generation of Gamma
-            # variables with non-integral shape parameters",
```
Comment (sweeneyde) [inline Lib/random.py]: This looks like a reference that should not be changed

Comment (skirpichev) [reply Lib/random.py]: Indeed,  thank you.

File: Doc/library/decimal.rst [hunk 2]
```diff
@@ -1352,7 +1352,7 @@ In addition to the three supplied contexts, new contexts can be created with the
       Return ``x`` to the power of ``y``, reduced modulo ``modulo`` if given.
 
       With two arguments, compute ``x**y``.  If ``x`` is negative then ``y``
-      must be integral.  The result will be inexact unless ``y`` is integral and
+      must be integer.  The result will be inexact unless ``y`` is integer and
```
Comment (mdickinson) [inline Doc/library/decimal.rst]: Here and in lots of other places in this PR, the new text isn't grammatically correct. The word "integer" is a noun, but here you're trying to use it as an adjective. "must be *an* integer" would work, but the existing "integral" is fine, too - I don't see a good reason to change this.

I'd strongly recommend splitting this PR up into two separate PRs: one for the `Integral` -> `Integer` change, and one for the mass of grammatical changes. Those two sets of changes need quite different review processes.

**Comment by skirpichev** [comment]
On Mon, Apr 26, 2021 at 12:20:04AM -0700, Mark Dickinson wrote:
>    @mdickinson commented on this pull request.
> 
>    ══════════════════════════════════════════════════════════════════════════
> 
>    In [1]Doc/library/decimal.rst:
> 
>  > @@ -1352,7 +1352,7 @@ In addition to the three supplied contexts, new contexts can be created with the
>         Return ``x`` to the power of ``y``, reduced modulo ``modulo`` if given.
> 
>         With two arguments, compute ``x**y``.  If ``x`` is negative then ``y``
>  -      must be integral.  The result will be inexact unless ``y`` is integral and
>  +      must be integer.  The result will be inexact unless ``y`` is integer and
> 
>    Here and in lots of other places in this PR, the new text isn't
>    grammatically correct. The word "integer" is a noun, but here you're
>    trying to use it as an adjective.

Well, it seems that Python comments invent a new noun, integral (as a
synonym for integer).  I did same, but invented an adjective)

But you are correct, of course.

>    I'd strongly recommend splitting this PR up into two separate PRs: one for
>    the Integral -> Integer change

I hope, I did all reversions.

**Pull request closed without merging**